app/helpers/request_info_helper.rb

```diff
@@ -23,20 +25,46 @@ def prov_grid_data(edit, vms, vm)
     {
       :headers  => prov_grid_vm_header(edit, clones, vms),
       :rows     => rows,
-      :selected => selected_item(edit).presence || none_index,
+      :selected => selected_vm(edit).presence || none_index,
       :recordId => edit[:req_id] || "new",
     }
   end
 
-  private
+  def prov_host_grid_data(edit, hosts)
+    none_index = '__HOST__NONE__'
+    rows = []
+    rows.push({:id => none_index, :clickable => true, :cells => none_cells(5)})
+    hosts.each do |row|
+      rows.push({:id => row.id.to_s, :clickable => true, :cells => prov_host_grid_cells(row, edit)})
+    end
+    {
+      :headers  => prov_grid_host_header(edit),
+      :rows     => rows,
+      :selected => selected_host(edit).presence || none_index,
+      :recordId => (edit && edit[:req_id]) || "new",
+    }
+  end
 
   def prov_grid_vm_header(edit, clones, vms)
     headers = []
     edit[:vm_columns].each_with_index do |h, index|
       item = {:text => edit[:vm_headers][h], :header_text => edit[:vm_headers][h]}
       if vms && clones.exclude?(edit[:wf].request_type)
         item[:sort_choice] = h
-        item[:sort_data] = sort_data(edit, index)
+        item[:sort_data] = sort_data(edit, index, 'vm')
+      end
+      headers.push(item)
+    end
+    headers
+  end
+
+  def prov_grid_host_header(edit)
+    headers = []
+    edit[:host_columns].each_with_index do |h, index|
+      item = {:text => edit[:host_headers][h], :header_text => edit[:host_headers][h]}
+      if edit
```

Comment:
When can this ever be falsey?  This block can only be called if `edit[:host_columns]` exists, which implies that `edit` also exists.

Reply:
In the old code, this was actually either edit or options, which i guess could have been nil? but here, it's passed in and assumed to exist.

Reply:
had made a bit of rework on this area, by passing the options and making checks.